Makefile

```diff
@@ -120,12 +120,22 @@ SOTNASSETS_SOURCES := $(shell find tools/sotn-assets -name '*.go')
 
 CHECK_FILES := $(shell cut -d' ' -f3 config/check.$(VERSION).sha)
 
+N_THREADS ?= $(shell nproc)
+
 .PHONY: build
 
 all: build check
+
+init: 
+	$(MAKE) clean
+	$(MAKE) extract -j $(N_THREADS)
+	$(MAKE) build -j $(N_THREADS)
+	$(MAKE) check
```

Comment:
the default target will run `build` and `check` and `check` _should_* have appropriate dependencies to only run after `build` is compete (it depends on all of the files it checks, which in turn depend on the targets to build those files), especially with a freshly cleaned build dir. 

\* this can fail when a checked file is present (previously built), but the target dependency graph isn't complete.